dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/BratWriter.java

```diff
@@ -610,13 +615,42 @@ private void writeRelationAnnotation(BratAnnotationDocument aDoc, FeatureStructu
         }
     }
 
+
+    
+    private BratTextAnnotation splitNewline(AnnotationFS aFS)
+    {
+
+        Pattern p = Pattern.compile("(.+?)(?:\\R|$)+", Pattern.DOTALL);
+        Matcher m = p.matcher(aFS.getCoveredText());
+        // counts the number of matches to initialize arrays sized
+        int i = 0;
+        while (m.find()) {
+            i++;
+        }
+        // initialize arrays
+        int[] begins = new int[i];
+        int[] ends = new int[i];
+        m = p.matcher(aFS.getCoveredText());
```

Comment:
Please avoid re-assigning variables. It makes the code more difficult to understand.